### PR TITLE
BFS Traversal of Binary Tree (in C++) and updated README

### PR DESCRIPTION
**Pull request by imsushant12**
I have implemented BFS traversal of Binary Tree with proper description, sample input/output and time/space complexity in C++ under GSSoC'21 

Related issue : https://github.com/Algo-Phantoms/Algo-Tree/issues/248

### CODE REVIEW

**Review by rish-singhal — COMMENTED** [review]
Add proper spacing after punctuations and operators.

**Comment by imsushant12** [comment]
> Add proper spacing after punctuations and operators.

Done, please review it!

File: Tree/readme.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ that, this property should be satisfied at every node in the tree.
 * Sorting the elements of binary search tree and many more
 
 ## Questions :
+* Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp)
```
Comment (rish-singhal) [inline Tree/readme.md:50]: isn't this line already there at Line 53?

File: Code/C++/binary_tree_BFS_traversal.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+{
+    Node *root = NULL;
+    root = create();
+    cout << "BFS Traversal of the tree is : ";
+    BFS_traversal(root);
+    return 0;
+}
+
+/*
+Time Complexity : O(n)
```
Comment (rish-singhal) [inline Code/C++/binary_tree_BFS_traversal.cpp:78]: describe n

File: Code/C++/binary_tree_BFS_traversal.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    tree->info = data;
+
+    cout << "Enter left child of " << data;
+    tree->left = create();
+
+    cout << "Enter right child of " << data;
+    tree->right = create();
+
+    return tree;
+};
```
Comment (rish-singhal) [inline Code/C++/binary_tree_BFS_traversal.cpp:32]: add extra line

File: Code/C++/binary_tree_BFS_traversal.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
+#include <bits/stdc++.h>
```
Comment (rish-singhal) [inline Code/C++/binary_tree_BFS_traversal.cpp:1]: add a generic description at the top. See other code samples in the repo for reference

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Tree/readme.md (hunk 1 above)
Comment (rish-singhal) [reply Tree/readme.md:50]: also make sure it's in alphabetical order.

File: Code/C++/binary_tree_BFS_traversal.cpp (hunk 4 above)
Comment (imsushant12) [reply Code/C++/binary_tree_BFS_traversal.cpp:6]: done, please review

File: Code/C++/binary_tree_BFS_traversal.cpp (hunk 3 above)
Comment (imsushant12) [reply Code/C++/binary_tree_BFS_traversal.cpp:37]: done

File: Code/C++/binary_tree_BFS_traversal.cpp (hunk 2 above)
Comment (imsushant12) [reply Code/C++/binary_tree_BFS_traversal.cpp]: done

File: Tree/readme.md (hunk 1 above)
Comment (imsushant12) [reply Tree/readme.md:50]: sorry I could not understand what you are saying

File: Tree/readme.md [hunk 5]
```diff
@@ -47,6 +47,7 @@ that, this property should be satisfied at every node in the tree.
 * Sorting the elements of binary search tree and many more
 
 ## Questions :
+* Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp) 
 * Check if a Binary Tree is a valid Binary Search Tree ----> [Java](/Code/Java/check_valid_BST.java)
 * Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp)
```
Comment (rish-singhal) [inline Tree/readme.md:52]: see it is already there... the line you added

File: Tree/readme.md [hunk 6]
```diff
@@ -55,3 +56,4 @@ that, this property should be satisfied at every node in the tree.
 * Spiral Traversal of Binary Tree ----> [C++](/Code/C++/spiral_traversal_of_binary_tree.cpp) 
 * Root to Leaf Path with Given Sum ----> [C++](/Code/C++/Root_to_leaf_path_with_given_sum.cpp)
 * Lowest common ancestor in a binary tree ----> [Python](/Code/Python/LCA_in_binary_tree.py)
+* BFS traversal on Binary Tree ----> [C++](/Code/C++/binary_tree_BFS_traversal.cpp)
```
Comment (rish-singhal) [inline Tree/readme.md:59]: also, please arrange these links in alphabetical order... i.e. 
1. BFS...
2. construction...
3.....
.. Searching
.... Spiral

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Tree/readme.md (hunk 5 above)
Comment (imsushant12) [reply Tree/readme.md]: Done

File: Tree/readme.md (hunk 6 above)
Comment (imsushant12) [reply Tree/readme.md]: Done sir

File: Code/C++/binary_tree_BFS_traversal.cpp [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+Enter left child of 40
+Enter data to be inserted or type -1 : -1
+Enter right child of 40
+Enter data to be inserted or type -1 : 50
+Enter left child of 50
+Enter data to be inserted or type -1 : -1
+Enter right child of 50
+Enter data to be inserted or type -1 : -1
+BFS Traversal of the tree is : 10 20 40 30 50
+
```
Comment (rish-singhal) [inline Code/C++/binary_tree_BFS_traversal.cpp:110]: add one more test case.

File: Code/C++/binary_tree_BFS_traversal.cpp [hunk 8]
```diff
@@ -0,0 +1,118 @@
+/*Binary tree is a data structure that has at most 2 child node i.e. left and right child
```
Comment (rish-singhal) [inline Code/C++/binary_tree_BFS_traversal.cpp:1]: also, update spacing here as 
/*
TEXT....
*/

not like 
/* TEXT */

for better indentation

**Review by rish-singhal — CHANGES_REQUESTED** [review]
these are final changes will merge after this ㊗️

File: Code/C++/binary_tree_BFS_traversal.cpp (hunk 8 above)
Comment (imsushant12) [reply Code/C++/binary_tree_BFS_traversal.cpp]: done sir, please review it

File: Code/C++/binary_tree_BFS_traversal.cpp (hunk 7 above)
Comment (imsushant12) [reply Code/C++/binary_tree_BFS_traversal.cpp:112]: done, please review.

**Review by rish-singhal — APPROVED** [review]

**Pull request merged**